packages/redlink-search/lib/SearchProvider.js

```diff
@@ -0,0 +1,47 @@
+import {
+	SearchProvider,
+	searchProviders,
+	SearchProviderMetadata,
+	SearchProviderRuntimeIntegration,
+	SearchProviderUi
+} from 'meteor/rocketchat:search';
+
+export const SETTING_URL = 'RedlinkSearchUrl';
+
+class RedlinkSearchProviderRuntimeIntegration extends SearchProviderRuntimeIntegration {
+
+}
+
+class RedlinkSearchProviderMetadata extends SearchProviderMetadata {
+	supportsPermissions() {
+		return true;
+	}
+
+	addSettings(section) {
+		section.add(SETTING_URL, '');
+	}
+
+	isConfigurationValid(settings, logger) {
+		logger.debug('Validating settings', settings);
+		if (!settings[SETTING_URL] || settings[SETTING_URL] === '') {
+			return false;
+		}
+	}
+}
+
+/*
+Don't know whether an own UI is really necessary
+ */
+class RedlinkSearchProviderUi extends SearchProviderUi {
+
+}
```

Comment:
If implemented what would this do beyond the settings exposed?

Reply:
@geekgonecrazy there are usecases where a search provider has own capabilities, such as filtering, facetting, more-like-this, ... 
A generic UI would have to be highly configurable in order to deal with all the different options - and it would be tricky to get all providers aligned.

Reply:
@graywolf336 thoughts? :top: this is going to be important across various parts of rocketchat apps.

Also we don't want to count on blaze.... as we might be switching frameworks.  So something agnostic would be best if this is needed.

Reply:
@geekgonecrazy one more thing about that: many external systems (which are to be integrated) already bring some sort of UI. For search, there are plenty of libs. None of them is written in blaze. This, a template which loads the other technology‘s UI onRendered is at least something to consider. As you said: also for the RC apps